### ISSUE TITLE
Warn when v-for with a Range is not a valid integer

### ISSUE THREAD

**Issue by Sand3r205** [vuejs/vue#10375]
### What problem does this feature solve?
When using the v-for directive with a Range and the number passed is not a valid integer (valid integer = an integer between 0 and 2^32-1), Vue.JS still tries to create an Array and then throws the error `[Vue warn]: Error in render: "RangeError: Invalid array length"`.

This happened to me while passing a computed property to the directive: `v-for="n in lists"` where `lists` is the computed property.

When the developer has multiple v-for directives, it is unclear where the error occurs, making the debugging process though.

If the mistake is caught by Vue before creating the Array, rendering the component shouldn't have to stop. Instead we can throw a warning and render an empty v-for directive. This would would make debugging easier, since other v-for directives in the same component would still render.

### What does the proposed API look like?
Add a check in /src/core/instance/render-helpers/render-list.js at line 22, checking if a valid number is:
* higher or equal to 0
* lower or equal to 2^32 - 1
* modules to 1 is equal to 0

```javascript
else if (typeof val === 'number') {
  if (val >= 0 && val <= 4294967295 && val % 1 === 0) {
    ret = new Array(val)
    for (i = 0; i < val; i++) {
      ret[i] = render(i + 1, i)
    }
  } else {
    warn (
      `Number passed to v-for directive not valid (expected valid integer), got ${val}`,
      this
    )
    ret = []
  }
}
```

**Comment by posva** [comment]
Maybe a dev warning could be fired instead of adding https://github.com/vuejs/vue/pull/10341 but at the same time, I don't understand how you find yourself writing `v-for="n in 10000000000` apart from testing purposes but it is a very unrealistic case. Let's wait for more feedback on this

FYI, the integer can also be writing in hex, making it more explicit that it is the upper bound of 4 bytes: `0xffffffff`

**Comment by Sand3r205** [comment]
Directly passing a number to the v-for directive is indeed a very straight forward case, however when passing a computed variable to the v-for directive, the number can be anything. In my case it was a calculation possibly returning a number with decimal points, which will also break new Array().

**Comment by posva** [comment]
Yeah, but that's different, as you should ensure the number is an integer in the computed property
